sphinx/transforms/__init__.py

```diff
@@ -397,6 +400,26 @@ def get_tokens(self, txtnodes):
             yield (texttype[notsmartquotable], txtnode.astext())
 
 
+class DoctreeReadEvent(SphinxTransform):
+    """Emit :event:`doctree-read` event."""
+    default_priority = 880
+
+    def apply(self):
+        # type: () -> None
+        self.app.emit('doctree-read', self.document)
+
+
+class Versioning(SphinxTransform):
+    """Prepare versioning."""
+    default_priority = 880
+
+    def apply(self):
+        # type: () -> None
+        if self.env.versioning_condition:
+            # add uids for versioning
+            versioning.prepare(self.document)
```

Comment:
Suggestion: this `Versioning` class and `sphinx.versioning.UIDTransform` seem to be unified into one.

https://github.com/sphinx-doc/sphinx/blob/b5ddc8c6977ac964e3ba4a2eb4647decdc742f4f/sphinx/versioning.py#L180-L184

Reply:
Absolutely. Updated.